File: iosApp/Packages/SwiftAudioStreamEx/Sources/SwiftAudioStreamEx/Models/PlayerControlsStyle.swift
import Foundation

/// Which transport controls are exposed on the lock screen / CarPlay / Control Center.
///
/// iOS surfaces a fixed transport layout that shows *either* next/prev *or* ±15s skip,
/// never both — `MPRemoteCommandCenter` doesn't expose a way to force both pairs on
/// the lock screen or CarPlay. So this enum is intentionally two-valued on iOS.
public enum PlayerControlsStyle: String, CaseIterable, Sendable {
    /// Previous / play-pause / next. Default.
    case skipTrack
    /// −15s / play-pause / +15s.
    case skipSeconds

    public init(rawValueOrDefault: String?) {
        self = PlayerControlsStyle(rawValue: rawValueOrDefault ?? "") ?? .skipTrack
    }
}

File: androidApp/core/model/src/main/java/com/grateful/deadly/core/model/PlayerControlsStyle.kt
package com.grateful.deadly.core.model

/**
 * Which transport controls are exposed on the notification, lock screen, and Android Auto.
 */
enum class PlayerControlsStyle(val label: String) {
    SKIP_TRACK("Tracks"),
    SKIP_SECONDS("10s"),
    BOTH("Both");

    companion object {
        fun fromString(value: String?): PlayerControlsStyle =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) } ?: SKIP_TRACK
    }
}
